app/controllers/site.rb

```diff
@@ -54,6 +54,28 @@ class Site < Base
 			end
 
 			namespace "/:site" do
+				get "/", params: :invite do
+					raise Pundit::NotAuthorizedError.new unless pundit_user
+					skip_authorization
```

Comment:
Instead of raising the error and then skipping other authorization, couldn't we just make this part of a policy that fails if there is no user?